### ISSUE TITLE
What is the problem TPOT score less than 0

### ISSUE THREAD

**Issue by gaochen219** [EpistasisLab/tpot#595]
I am new to TPOT, am trying it and get the .py file as below. But the Score is negative. I thought the score shouldn't be less than 0. What is the issue?

Sometimes I also saw my "internal best CV score" from TPOT.score bigger than 1. I thought cross validation score should be between 0-1 as accuracy %. How could I understand thr result? Thanks!

# Score on the training set was:-0.0641924968157452
exported_pipeline = make_pipeline(
    make_union(
        StackingEstimator(estimator=DecisionTreeRegressor(max_depth=3, min_samples_leaf=20, min_samples_split=19)),
        FunctionTransformer(copy)
    ),
    GradientBoostingRegressor(alpha=0.75, learning_rate=0.5, loss="huber", max_depth=8, max_features=0.9000000000000001, min_samples_leaf=11, min_samples_split=8, n_estimators=100, subsample=0.9500000000000001)
)

exported_pipeline.fit(training_features, training_target)
results = exported_pipeline.predict(testing_features)

**Comment by weixuanfu** [comment]
Hmm, for regression problem, R^2 score may be negative (it need not actually be the square of a quantity R). Check this [link](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html#sklearn.metrics.r2_score) for more details. Maybe there is a issue about overfitting in training dataset.

**Comment by gaochen219** [comment]
Thank you! It is a regression. Any suggestion on how to solve the overfitting issue? Thanks!

Another questioon, regression again, best internal CV score is bigger than 1 (1.8, 2.5....) - is it reasonable? Thanks a lot!

**Comment by weixuanfu** [comment]
The default CV score in TPOT was based on [mean_squared_error](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html#sklearn.metrics.mean_squared_error). I cannot tell whether it is reasonable without knowing details of your data, like sample size and how bias the dataset is. You may try to tune `subsample` parameter in TPOT to solve this overfitting issue.

**Comment by gaochen219** [comment]
Thank you very much!

**Comment by gaochen219** [comment]
@weixuanfu Thanks again for your help. I realized that the score error may come from the data bias (a ton of 0 in the Y). Therefore I have another question here: https://github.com/rhiever/tpot/issues/598. Would you help me take a look at it? Thanks!

**Comment by sergiolucero** [comment]
Actually the answer is to look at your
    exported_pipeline.scoring_function
which, at least in my case, turns out to be 'neg_mean_squared_error', so your result is MSE=6.42%

**Comment by Vamsi5689** [comment]
@weixuanfu I tried with 0.1 to 0.9 subsample values but I can't prevent any other suggestions

**Comment by weixuanfu** [comment]
@Vamsi5689 Could you please let me know more details about the issue, like a demo?

**Comment by Vamsi5689** [comment]
![image](https://user-images.githubusercontent.com/33513617/61049959-77023000-a403-11e9-85c4-c4b3b814f7a8.png)
I am getting negative values in cv and I tried changes in subsample but the same thing happened again and again

**Comment by weixuanfu** [comment]
Which scoring function was used in TPOT? If it is default scoring function, then scoring is 'neg_mean_squared_error' and cv scores should be a negative value.  Please check [this link](https://scikit-learn.org/stable/modules/model_evaluation.html#common-cases-predefined-values)
